client/my-sites/plugins/with-plugin-redirect/index.js

```diff
@@ -0,0 +1,190 @@
+				isInstallingPlugin: isPluginActionInProgress( state, siteId, pluginSlug, INSTALL_PLUGIN ),
+				hasPluginBeenInstalled: isPluginActionCompleted(
+					state,
+					siteId,
+					pluginSlug,
+					INSTALL_PLUGIN
+				),
+				isAtomic: isSiteAutomatedTransfer( state, siteId ),
+				isJetpack: isJetpackSite( state, siteId ),
+				isSiteConnected: getSiteConnectionStatus( state, siteId ),
+				redirectUrl: redirectHandler?.getUrl( state, siteId ),
+			};
+		} );
+
+		/*
+		 * Define whether the site should redirect,
+		 * once the action is done (install plugin or transferring site).
+		 * It needs to detect, store nand compare previous state
+		 * of the installing/transferring process.
+		 */
+		useEffect( () => {
+			if ( ! isSiteConnected ) {
+				return;
+			}
+
+			// Jetpack (installing-plugin): Flag if the site is installing a plugin.
+			if (
+				isJetpack &&
+				isInstallingPlugin &&
+				! hasPluginBeenInstalled &&
+				( prevProcessingState === 'init' || prevProcessingState === 'done' )
+			) {
+				setPrevProcessingState( 'installing-plugin' );
+			}
+
+			// Jetpack (plugin-installed): Flag if the site has intalled a plugin.
+			if (
+				isJetpack &&
+				! isInstallingPlugin &&
+				hasPluginBeenInstalled &&
+				prevProcessingState === 'installing-plugin'
+			) {
+				setPrevProcessingState( 'plugin-installed' );
+			}
+
+			// Atomic (complete): flag if the site was being transferred.
+			if (
+				isAtomic &&
+				transferringStatus === transferStates.COMPLETE &&
+				prevProcessingState === 'init'
+			) {
+				setPrevProcessingState( transferStates.COMPLETE );
+			}
+
+			if (
+				( isJetpack && prevProcessingState === 'plugin-installed' ) || // <- Jetpack site.
+				( isAtomic && prevProcessingState === transferStates.COMPLETE ) // <- Atomic site.
+			) {
+				return setShouldRedirect( redirectUrl );
+			}
+
+			setShouldRedirect( false );
+		}, [
+			isInstallingPlugin,
+			hasPluginBeenInstalled,
+			redirectUrl,
+			transferringStatus,
+			isJetpack,
+			isAtomic,
+			isSiteConnected,
+			prevProcessingState,
+		] );
+
+		// Perform the client redirect,
+		// cancelable by clickin on the cancel button.
+		useEffect( () => {
+			if ( ! shouldRedirect ) {
+				return;
+			}
+
+			// For Atomic sites, we shoulnd't wait since
+			// there is not a Notice about installing the plugin.
+			const pluginInstalledNoticeTime = transferringStatus === transferStates.COMPLETE ? 100 : 3000;
+
+			let timerId = setTimeout( () => {
+				dispatch( removeNotice( 'plugin-notice' ) );
+
+				// re-use same timer ID to redirect the client.
+				timerId = setTimeout( () => {
+					if ( ! window?.top.location.href ) {
+						return;
+					}
+
+					return ( window.location.href = shouldRedirect );
+				}, 6000 );
+
+				const redirectMessage =
+					redirectHandler.message ||
+					translate( 'Redirecting to setup %{pluginName} in five seconds.', {
+						args: {
+							pluginName: redirectHandler.name,
```

Comment:
Should we add the number of seconds as an argument too?

Reply:
why should we do that, considering that the number of seconds is not a variable?